challenge_serving_drinks/src/challenge_serving_drinks/sd_states.py

```diff
@@ -0,0 +1,483 @@
+                    "Would you tell me which is the unavailable drink?",
+                    "Please tell me which drink is unavailable",
+                ]),
+                block=True
+            )
+
+            # Try to get the answer
+            try:
+                speech_result = self._query_drink(self._max_queries_per_try)
+            except TimeoutException:
+                continue
+
+            # Ask for confirmation
+            self._robot.speech.speak(
+                "I understood that the {} is unavailable, is that correct?".format(speech_result.semantics)
+            )
+            if not self._confirm():
+                continue
+
+            # Store the type in the designator
+            self._unavailable_drink_designator.write(str(speech_result.semantics))
+            rospy.loginfo("Unavailable drink: {}".format(self._unavailable_drink_designator.resolve()))
+
+            return "succeeded"
+
+        self._robot.speech.speak("I am sorry but I cannot understand you. I will continue with my tasks", block=False)
+        self._robot.head.cancel_goal()
+        rospy.loginfo("Unavailable drink still unknown")
+        return "failed"
+
+    def _query_drink(self, max_tries):
+        """
+        Queries the HMI server for <max_tries> times until a result has been obtained
+        :return: (HMIResult) (N.B.: the 'semantics' attribute should be a dict with 'beverage' key)
+        :raises: (TimeoutException) if robot didn't hear anything
+        """
+        attempt = 0
+        while not rospy.is_shutdown():
+            attempt += 1
+            try:
+                speech_result = self._robot.hmi.query(description="Which is the unavailable beverage?",
+                                                      grammar=self._drinks_grammar,
+                                                      target=self._drinks_target)
+                return speech_result
+            except TimeoutException:
+                if attempt >= max_tries:
+                    raise  # Re-raise
+
+            # We only end up here if the exception was caught
+            self._robot.speech.speak(random.choice([
+                "I'm sorry, can you repeat",
+                "Please repeat the drink, I didn't hear you",
+                "I didn't get the drink, can you repeat it",
+                "Please speak up, as I didn't hear what you just said",
+            ]))
+
+    def _confirm(self):
+        """
+        Queries the HMI server for confirmation
+        :return: (bool) whether the understood beverage was correct
+        """
+        try:
+            speech_result = self._robot.hmi.query(description="Is this correct?", grammar="T[True] -> yes;"
+                                                                                          "T[False] -> no", target="T")
+        except TimeoutException:
+            return False
+
+        return speech_result.semantics
```

Comment:
Function should return a bool according to docstring. This is a string.

Reply:
Did a little test:
```python
    def _confirm(_robot):
        """
        Queries the HMI server for confirmation
        :return: (bool) whether the understood beverage was correct
        """
        try:
            speech_result = _robot.hmi.query(description="Is this correct?", grammar="T[True] -> yes;"
                                                                                          "T[False] -> no", target="T")
        except TimeoutException:
            return False

        print("Speech semantics: {} of type {}".format(speech_result.semantics, type(speech_result.semantics)))
        return speech_result.semantics

    _confirm(_robot)
```
results in 
```
Speech semantics: True of type <type 'bool'>
```